File: JoyReactor.Accordion.WebAPI/Models/Responses/CrawlerTaskResponse.cs
using JoyReactor.Accordion.Logic.ApiClient.Models;
using JoyReactor.Accordion.Logic.Database.Sql.Entities;

namespace JoyReactor.Accordion.WebAPI.Models.Responses;

public record CrawlerTaskResponse
{
    public CrawlerTaskResponse() { }
    public CrawlerTaskResponse(CrawlerTask crawlerTask)
    {
        Id = crawlerTask.Id;
        Tag = new ParsedTagThinResponse(crawlerTask.Tag);
        PostLineType = crawlerTask.PostLineType;
        PageCurrent = crawlerTask.PageCurrent;
        PageLast = crawlerTask.PageLast;        
        StartedAt = crawlerTask.StartedAt;
        FinishedAt = crawlerTask.FinishedAt;
        CreatedAt = crawlerTask.CreatedAt;
        UpdatedAt = crawlerTask.UpdatedAt;
    }

    public Guid Id { get; set; }
    public ParsedTagThinResponse Tag { get; set; }

    public PostLineType PostLineType { get; set; }

    public int PageCurrent { get; set; }
    public int? PageLast { get; set; }

    public DateTime? StartedAt { get; set; }
    public DateTime? FinishedAt { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: JoyReactor.Accordion.Web/src/app/statistics/crawler-task-service/crawler-task-response.ts
export interface CrawlerTaskResponse {
  id: string,
  tag: ParsedTagThinResponse,
  postLineType: number
  pageFrom?: number
  pageTo?: number,
  pageCurrent?: number,
  isIndefinite: boolean
  isCompleted: boolean,
  startedAt?: Date,
  finishedAt?: Date,
  createdAt: Date,
  updatedAt: Date,
}

export interface ParsedTagThinResponse {
  id: string,
  numberId: number,
  name: string,
}